src/Vimeo/Vimeo.php

```diff
@@ -256,63 +256,92 @@ public function buildAuthorizationEndpoint($redirect_uri, $scope = 'public', $st
     }
 
     /**
-     * Upload a file. This should be used to upload a local file.
-     * If you want a form for your site to upload direct to Vimeo,
-     * you should look at the POST /me/videos endpoint.
+     * Upload a file.
      *
+     * This should be used to upload a local file. If you want a form for your site to upload direct to Vimeo, you
+     * should look at the `POST /me/videos` endpoint.
+     *
+     * @link https://developer.vimeo.com/api/endpoints/videos#POST/users/{user_id}/videos
      * @param string $file_path Path to the video file to upload.
-     * @param string|null $machine_id
+     * @param array $params Parameters to send when creating a new video (name, privacy restrictions, etc.).
      * @throws VimeoUploadException
      * @return string Video URI
      */
-    public function upload($file_path, $machine_id = null)
+    public function upload($file_path, array $params = [])
     {
         // Validate that our file is real.
         if (!is_file($file_path)) {
             throw new VimeoUploadException('Unable to locate file to upload.');
         }
 
-        // Begin the upload request by getting a ticket
-        $ticket_args = array('type' => 'streaming');
-        if ($machine_id !== null) {
-            $ticket_args['machine_id'] = $machine_id;
+        $file_size = filesize($file_path);
+
+        // If the user does not have enough free space in their quota to upload this, then don't.
+        $response = $this->request('/me', array('fields' => 'upload_quota.space.free'), 'GET');
+        if ($response['status'] !== 200) {
+            $error = !empty($response['body']['error']) ? ' [' . $response['body']['error'] . ']' : '';
+            throw new VimeoUploadException('Unable to pull the users upload quota.' . $error);
+        } elseif ($file_size > $response['body']['upload_quota']['space']['free']) {
+            throw new VimeoUploadException('User does not have any more free space to upload this video.');
+        }
+
+        // Ignore any specified upload approach and size.
+        $params['upload']['approach'] = 'tus';
```

Comment:
How come 'tus' is the only approach supported for upload? Did I miss something?

Reply:
AFAIK, that's going to be the default upload approach for all these SDKs.

Reply:
The upload function should upload via tus. If someone wants to use pull or POST they need to make the API call themselves via the request method.